### DISCUSSION TITLE
TNS_MAX_CURSORS_TO_CLOSE: Why is it limited to 500 ? Anyway to increase it's value ?

### DISCUSSION THREAD

**Discussion by Alaeddine22** [oracle/python-oracledb#372] [Q&A] [answered]
Hello,

recently i bumped into an error message from oracledb saying `DPY-5008: internal error: attempt to close more than 500 cursors`
Searching for the reason, i found that the threshold of TNS_MAX_CURSORS_TO_CLOSE is the cause.
I didn't find a direct way to increase its value of 500.

i doubt that there might be a logical reason for this limit but i didn't find an explanation on the web..

Questions:

1. What it the reason for the limit of 500 cursors to close?
2. Anyway to increase this value ?

Thanks in advance

**Comment by anthony-tuininga** [comment, marked as answer]
This limitation was removed in version 2.1. It is now limited to the maximum number of open cursors allowed by the database -- since any attempt to use more than that will fail anyway! My suggestion: upgrade. :)

**Comment by Alaeddine22** [comment]
All clear now.
Thank you for the quick response!
